models.py

```diff
@@ -232,9 +232,37 @@ def location_network(rnn_state, sampling, std=params['std'],
     # core network
     # TODO: parameterize rnn_cell, i.e. implement split cell from paper
     with tf.variable_scope('core_network', reuse=tf.AUTO_REUSE):
-        rnn_cell = tf.nn.rnn_cell.LSTMCell(params['core_size'])
-        state = rnn_cell.zero_state(batch_size, tf.float32)
-
+        if params['core_type'] == 'LSTM':
+            rnn_cell = tf.nn.rnn_cell.LSTMCell(params['core_size'])
+            state = rnn_cell.zero_state(batch_size, tf.float32)
+
+        else:
+            def rnn_cell(glimpse, state): 
+                
+                dense_state = tf.layers.dense(
+                    inputs=state,
+                    units=params['core_size'],
+                    activation=None)
+            
+                dense_glimpse = tf.layers.dense(
+                    inputs=glimpse,
+                    units=params['glimpse_out_size'],
+                    activation=None)
+                
+                dstate_plus_dglimpse = tf.add(
+                        x=dense_state,
+                        y=dense_glimpse)
+                
+                rnn_cell = tf.contrib.rnn.BasicRNNCell(
+                    num_units=params['core_size'],
+                    activation=tf.nn.relu)
+    
+                output, state = tf.dynamic_rnn(
+                        cell=rnn_cell,
+                        inputs=dstate_plus_dglimpse)
```

Comment:
No need to define `rnn_cell` and use `dynamic_rnn` here.  I'll explain more in person, but the basic idea is that what I've done with a `tf.while_loop` later in the code is what would otherwise be carried out by `dynamic_rnn`

So, this `def rnn_cell` only needs to take in `glimpse` and `state` and output a pair of `output` and `state` for the next step.  [More on that below.]

So, what should go in place of this block of code is a `ReLu` dense layer on the sum you took above (corresponding to `Rect(Linear(ht-1) + Linear(gt))` in the paper).  I'd also use `params['core_size']` for its size.  In principle, we could vary these, but I don't see a need to right now.